### PR TITLE
Brennan fixes

### PR DESCRIPTION
**Pull request by SeanNobel**

### CODE REVIEW

File: speech_decoding/constants.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import torch, random
+import numpy as np
+
+# assert torch.cuda.is_available(), "Training without GPU is not supported."
+DEVICE = torch.device("cuda" if torch.cuda.is_available() else "cpu")
```
Comment (Kaixhin) [inline speech_decoding/constants.py]: This really shouldn't be hardcoded, unless for some reason it's not possible to run the code without a GPU (being too slow doesn't count). If you need to disable the GPU then that should be a flag.

File: speech_decoding/constants.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+BAR_FORMAT = "{desc:<5.5}{percentage:3.0f}%|{bar:10}{r_bar}"
+
+# NOTE: I'm using GPU on my machine so heavily on other project that I can't use
+# device = "cpu"
+
+BRAIN_RESAMPLE_RATE = 120
```
Comment (Kaixhin) [inline speech_decoding/constants.py:4]: Why were these added to `constants.py` instead of being in the config as before?

File: speech_decoding/constants.py (hunk 1 above)
Comment (SeanNobel) [reply speech_decoding/constants.py]: Understand. I've put back device and bar_format to how it was in `tidy`.

File: speech_decoding/constants.py (hunk 2 above)
Comment (SeanNobel) [reply speech_decoding/constants.py:4]: I thought those that are not hyperparameters and shouldn't be changed could be in `constants.py`?

Comment (Kaixhin) [reply speech_decoding/constants.py:4]: I see. I think in that case it should be OK (Python doesn't have any recommended way of dealing with this and `constants.py` is one solution) - but if the code can be run on CPU then definitely `DEVICE` should not be there and code should be device-agnostic.

**Comment by SeanNobel** [comment]
@Kaixhin  After doing some visualizations with Brennan, I think the dataset class is not doing anything really wrong. ([except that I cannot find the relative relationship of word and EEG onsets](https://github.com/SeanNobel/speech-decoding/blob/brennan-fixes/speech_decoding/dataclass/brennan2018.py#L47-L51))
https://github.com/SeanNobel/speech-decoding/blob/brennan-fixes/notebooks/test_brennan2018.ipynb

But, 
- Looking at this subject 1, channel  48 is pretty noisy and probably should be removed in usual Neuroscience research. There should be channels like this in other subjects.
- Audio embeddings look so similar from human eyes. (probably caused by the green feature around 700)

Will be posting the loss curve for current version later.

**Comment by Kaixhin** [comment]
> @Kaixhin After doing some visualizations with Brennan, I think the dataset class is not doing anything really wrong. ([except that I cannot find the relative relationship of word and EEG onsets](https://github.com/SeanNobel/speech-decoding/blob/brennan-fixes/speech_decoding/dataclass/brennan2018.py#L47-L51)) https://github.com/SeanNobel/speech-decoding/blob/brennan-fixes/notebooks/test_brennan2018.ipynb

Thanks for checking, good to note, but also I think your assumptions seem reasonable.

> * Looking at this subject 1, channel  48 is pretty noisy and probably should be removed in usual Neuroscience research. There should be channels like this in other subjects.

To a non-expert eye (me), I can't tell any channels are good or bad 😕 Unless the paper said otherwise, I think they kept all channels, and expect the DL model to just ignore noisy ones? Certainly a subject-specific layer could do that.

> * Audio embeddings look so similar from human eyes. (probably caused by the green feature around 700)

Looking very carefully, the horizontal bands seem similar (freq constant over time, makes sense), but I can see differences in the vertical bands, so I think it's OK.

**Comment by SeanNobel** [comment]
The scale of channel 48 of subject 1 after segmenting is thousands, while others are tens.
Due to that channel 48 is clipping so much after clamping. (we could avoid this by scaling each channel independently)

**Comment by Kaixhin** [comment]
> (we could avoid this by scaling each channel independently)

Ah, this seems like it would be normal practice from an ML perspective. If the paper doesn't explicitly say otherwise, probably we should try this?

**Comment by SeanNobel** [comment]
CodeRabbit test
